### PR TITLE
Fix alias check

### PR DESCRIPTION
**Pull request by Aiee**
1. Check for MATCH sentence alias when edgeProperty expressions are used.
    ```
    (root@nebula) [nba]> match p = (v:player{name:"Tony Parker"})-[e]->(v2) where v._src>0 return e
    [ERROR (-12)]: SemanticError: Vertex `v' does not have the src attribute
    ```
2. Add more intuitive error messages.
    ```
    (root@nebula) [nba]> match p = (v:player{name:"Tony Parker"})-[e]->(v2) return e._src
    [ERROR (-12)]: SemanticError: To get the srcID of the edge, use src(e)
    ```

Fix https://jira.nebula-graph.io/browse/NG-529

### CODE REVIEW

File: src/validator/MatchValidator.cpp [hunk 1]
```diff
@@ -755,6 +750,103 @@ Status MatchValidator::validateYield(YieldClauseContext &yieldCtx) const {
...
+    }
+    return Status::OK();
+}
+
+StatusOr<AliasType> MatchValidator::getAliasType(
+    const std::unordered_map<std::string, AliasType> *aliasesUsed,
+    const std::string *name) const {
+    NG_RETURN_IF_ERROR(isAliasDefined(aliasesUsed, name));
+    auto iter = aliasesUsed->find(*name);
+    return iter->second;
```
Comment (yixinglu) [inline src/validator/MatchValidator.cpp:766]: Could the `iter` not be the `aliasesUsed.end()` ? If so, should `DCHECK` it.

Comment (Aiee) [reply src/validator/MatchValidator.cpp:766]: `isAliasDefined()` has ensured the presence of `name`, so `iter` will always point to a valid position.

**Review by yixinglu — DISMISSED** [review]

File: src/validator/MatchValidator.cpp [hunk 2]
```diff
@@ -755,6 +749,127 @@ Status MatchValidator::validateYield(YieldClauseContext &yieldCtx) const {
...
+    AliasType aliasType = AliasType::kDefault;
+
+    switch (kind) {
+        case Expression::Kind::kLabel: {
+            auto name = static_cast<const LabelExpression *>(refExpr)->name();
+            auto res = getAliasType(aliasesUsed, name);
+            if (!res.ok()) {
+                return res.status();
+            }
+            return Status::OK();
```
Comment (yixinglu) [inline src/validator/MatchValidator.cpp:776]: return getAliasType(aliasesUsed, name);

File: src/validator/MatchValidator.cpp [hunk 3]
```diff
@@ -755,6 +749,127 @@ Status MatchValidator::validateYield(YieldClauseContext &yieldCtx) const {
...
+        }
+        case Expression::Kind::kLabelAttribute: {
+            auto name = static_cast<const LabelAttributeExpression *>(refExpr)->left()->name();
+            auto res = getAliasType(aliasesUsed, name);
+            if (!res.ok()) {
+                return res.status();
+            }
+            return Status::OK();
```
Comment (yixinglu) [inline src/validator/MatchValidator.cpp:784]: ditto

File: src/validator/MatchValidator.cpp [hunk 4]
```diff
@@ -755,6 +749,127 @@ Status MatchValidator::validateYield(YieldClauseContext &yieldCtx) const {
...
+        }
+        case Expression::Kind::kEdgeSrc: {
+            auto name = static_cast<const EdgeSrcIdExpression *>(refExpr)->sym();
+            auto res = getAliasType(aliasesUsed, name);
+            if (!res.ok()) {
+                return res.status();
+            }
```
Comment (yixinglu) [inline src/validator/MatchValidator.cpp:791]: NG_RETURN_IF_ERROR(res);

File: src/validator/test/MatchValidatorTest.cpp [hunk 5]
```diff
@@ -516,5 +516,42 @@ TEST_F(MatchValidatorTest, with) {
...
+    }
+    // incorrect alias type in return clause
+    {
+        std::string query = "MATCH p = (v :person{name:\"Tim Duncan\"})-[e]-(v2) "
+                            "RETURN p._rank";
+        auto result = checkResult(query);
+        EXPECT_EQ(std::string(result.message()),
+                  "SemanticError: Path `p' does not have the ranking attribute");
+    }
+}
```
Comment (yixinglu) [inline src/validator/test/MatchValidatorTest.cpp:554]: Prefer more test cases for scenarios in `checkAlias`.

File: src/validator/MatchValidator.cpp [hunk 6]
```diff
@@ -754,6 +748,113 @@ Status MatchValidator::validateYield(YieldClauseContext &yieldCtx) const {
...
+            if (aliasType == AliasType::kNode) {
+                return Status::SemanticError("Vertex `%s' does not have the dst attribute",
+                                             name->c_str());
+            } else if (aliasType == AliasType::kEdge) {
+                return Status::SemanticError("To get the dst vid of the edge, use dst(%s)",
+                                             name->c_str());
+            } else if (aliasType == AliasType::kPath) {
+                return Status::SemanticError("To get the end node of the path, use endNode(%s)",
+                                             name->c_str());
+            }
```
Comment (yixinglu) [inline src/validator/MatchValidator.cpp:811]: It's better to use `switch..case` for these branches.

**Review by yixinglu — APPROVED** [review]

**Review by jievince — APPROVED** [review]

**Review by czpmango — APPROVED** [review]

**Pull request merged**